File: Joanna_Laurence/src/main/java/org/uva/sea/ql/evaluate/SymbolTable.java
package org.uva.sea.ql.evaluate;

import java.util.HashMap;

public class SymbolTable {

    private HashMap<String, Value> symbolTable = new HashMap<>();

    /**
     * Add or update symbol table
     *
     * @param name Name of the symbol
     * @param val  Value of the symbol
     */
    public void addOrUpdateValue(String name, Value val) {
        symbolTable.put(name, val);
    }

    /**
     * @param name Name of the symbol
     * @return Value of the symbol or null on error
     */
    public void getValue(String name) {
        symbolTable.get(name);
    }
}

File: DogeQL/src/main/kotlin/symbol/SymbolTable.kt
package symbol

import common.Identifier

class SymbolTable {

    val table = HashMap<Identifier, SymbolValue>()

    fun lookUp(identifier: Identifier): SymbolValue? {
        return table[identifier]
    }

    fun register(identifier: Identifier, symbol: SymbolValue) {
        table[identifier] = symbol
    }

}